sklearn/utils/multiclass.py

```diff
@@ -229,6 +229,26 @@ def is_multilabel(y):
     """
     return is_label_indicator_matrix(y) or is_sequence_of_sequences(y)
 
+def assert_non_regression_targets(y):
+    """Determine if target `y` is of non regression type
+
+    For the moment, the following (as defined in type_of_target) are allowed:
+        - 'binary', 'multiclass', 'multiclass-multioutput', 
+        'multilabel-indicator', 'multilabel-sequences'
+        - 'continuous', if only there are only two unique values
+
+    Parameters
+    ----------
+    y : array-like
+    """
+    y_type = type_of_target(y)
+    if y_type not in ['binary', 'multiclass', 'multiclass-multioutput', 
+            'multilabel-indicator', 'multilabel-sequences']:
+        if not y_type is 'continuous':
+            raise ValueError("Unknown label type: %r" % y)
+        elif len(np.unique(y)) > 2:
```

Comment:
When y is centered (in a test for RandomizedLogisticRegression), [0,1] becomes [-0.5, 0.5], therefore is continuous.
This is a bit arbitrary... Maybe it's better to change that test.